# Loading and Exploring the Paper 1 Microbiome Dataset

In [ ]:
import pandas as pd
import os

base = r"C:\Users\rtajz\Downloads\dietstudy_analyses-master\dietstudy_analyses-master\data"

print("Base exists:", os.path.exists(base))

microbiome = pd.read_csv(
    f"{base}/microbiome/processed_sample/taxonomy_counts_s.txt",
    sep="\t",
    index_col=0
)

sample_meta = pd.read_csv(
    f"{base}/maps/food_map.txt",
    sep="\t"
).rename(columns={"#SampleID": "SampleID"})

diet_raw = pd.read_csv(
    f"{base}/diet/processed_food/dhydrt.txt",
    sep="\t",
    index_col=0
)

diet_decay = pd.read_csv(
    f"{base}/diet/processed_food/masterdecaydiet.txt",
    sep="\t",
    index_col=0
)

nutrition_totals = pd.read_csv(
    f"{base}/diet/nutrition_totals.txt",
    sep="\t",
    index_col=0
)

microbiome_clr = pd.read_csv(
    f"{base}/microbiome/processed_sample/taxonomy_clr_s.txt",
    sep="\t",
    index_col=0
)

diet_fiber = pd.read_csv(
    f"{base}/diet/diet.fiber.txt",
    sep="\t",
    index_col=0
)

fiber_avg = pd.read_csv(
    f"{base}/diet/fiber/fiber.smry.otu.txt",
    sep="\t",
    index_col=0
)

micro_avg_counts = pd.read_csv(
    f"{base}/microbiome/processed_average/UN_taxonomy_counts_s.txt",
    sep="\t",
    index_col=0
)

micro_avg_clr = pd.read_csv(
    f"{base}/microbiome/processed_average/UN_taxonomy_clr_s.txt",
    sep="\t",
    index_col=0
)

micro_avg_norm = pd.read_csv(
    f"{base}/microbiome/processed_average/UN_taxonomy_norm_s.txt",
    sep="\t",
    index_col=0
)

print("=== microbiome ===")
print(microbiome.shape)
print(microbiome.head())

print("=== sample_meta ===")
print(sample_meta.shape)
print(sample_meta.head())

print("=== diet_raw ===")
print(diet_raw.shape)
print(diet_raw.head())

print("=== diet_decay ===")
print(diet_decay.shape)
print(diet_decay.head())

print("=== nutrition_totals ===")
print(nutrition_totals.shape)
print(nutrition_totals.head())

print("=== microbiome_clr ===")
print(microbiome_clr.shape)
print(microbiome_clr.head())

print("=== diet_fiber ===")
print(diet_fiber.shape)
print(diet_fiber.head())

print("=== fiber_avg ===")
print(fiber_avg.shape)
print(fiber_avg.head())

print("=== micro_avg_counts ===")
print(micro_avg_counts.shape)
print(micro_avg_counts.head())

print("=== micro_avg_clr ===")
print(micro_avg_clr.shape)
print(micro_avg_clr.head())

print("=== micro_avg_norm ===")
print(micro_avg_norm.shape)
print(micro_avg_norm.head())

microbiome.to_csv("microbiome.csv")
sample_meta.to_csv("sample_meta.csv", index=False)
diet_raw.to_csv("diet_raw.csv")
diet_decay.to_csv("diet_decay.csv")
nutrition_totals.to_csv("nutrition_totals.csv")
microbiome_clr.to_csv("microbiome_clr.csv")
diet_fiber.to_csv("diet_fiber.csv")
fiber_avg.to_csv("fiber_avg.csv")
micro_avg_counts.to_csv("micro_avg_counts.csv")
micro_avg_clr.to_csv("micro_avg_clr.csv")
micro_avg_norm.to_csv("micro_avg_norm.csv")

print("Done! CSV files were saved.")

Base exists: True
=== microbiome ===
(223, 483)
                                                    MCT.f.0002  MCT.f.0003  \
#taxonomy                                                                    
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...       79055       33364   
k__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...        2421       16944   
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...       39779        9983   
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...           1           5   
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...       18393         561   

                                                    MCT.f.0004  MCT.f.0005  \
#taxonomy                                                                    
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...       64027       70720   
k__Bacteria;p__Firmicutes;c__Clostridia;o__Clos...       43511       18335   
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__...       31539       44880   
k__Bacteria;p__

# Profiling the Paper 1 Gut Microbiome Abundance Dataset

In [5]:
import pandas as pd
import numpy as np

microbiome = pd.read_csv("microbiome.csv", index_col=0)

print("Shape:", microbiome.shape)
print("Number of microbial features:", microbiome.shape[0])
print("Number of samples:", microbiome.shape[1])

print("\nFirst rows:")
display(microbiome.head())

print("\nBasic statistics:")
display(microbiome.describe())

Shape: (223, 483)
Number of microbial features: 223
Number of samples: 483

First rows:


,MCT.f.0002,MCT.f.0003,MCT.f.0004,MCT.f.0005,MCT.f.0006,MCT.f.0007,MCT.f.0008,MCT.f.0009,MCT.f.0010,MCT.f.0011,...,MCT.f.0618,MCT.f.0619,MCT.f.0620,MCT.f.0621,MCT.f.0622,MCT.f.0623,MCT.f.0624,MCT.f.0625,MCT.f.0627,MCT.f.0628
#taxonomy,,,,,,,,,,,,,,,,,,,,,
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;NA,79055,33364,64027,70720,63361,58636,73520,69225,58080,75592,...,116037,93723,105413,100387,97278,117400,123705,96470,117867,93290
k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii,2421,16944,43511,18335,36493,43987,15944,20583,5493,23064,...,11,62,3,5,23,2,7,18,2206,5
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis,39779,9983,31539,44880,29901,31975,41607,39981,32774,41183,...,25278,26608,24590,26421,24128,23532,20010,22195,9340,21397
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__Prevotella copri,1,5,4,4,7,2,13,3,9,4,...,7,10,1,0,184,2,2,6,142,1
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__Alistipes putredinis,18393,561,11742,15831,14319,16541,8328,10082,40723,16670,...,11,52,6,9,13,5,10,9,1246,4



Basic statistics:


,MCT.f.0002,MCT.f.0003,MCT.f.0004,MCT.f.0005,MCT.f.0006,MCT.f.0007,MCT.f.0008,MCT.f.0009,MCT.f.0010,MCT.f.0011,...,MCT.f.0618,MCT.f.0619,MCT.f.0620,MCT.f.0621,MCT.f.0622,MCT.f.0623,MCT.f.0624,MCT.f.0625,MCT.f.0627,MCT.f.0628
count,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,...,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000,223.000000
mean,1196.116592,1196.130045,1196.143498,1196.085202,1196.107623,1196.098655,1196.098655,1196.121076,1196.116592,1196.076233,...,1196.085202,1196.107623,1196.125561,1196.134529,1196.161435,1196.071749,1196.139013,1196.147982,1196.107623,1196.076233
std,6292.628430,3929.367321,5764.614294,6018.009459,5532.632866,5599.856620,5909.878262,5687.239997,5478.451081,6156.212415,...,9425.191368,9155.216598,9278.133530,8784.064237,9080.516936,9225.165606,9218.804988,8983.242874,8727.492092,9334.324673
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,11.000000,13.000000,18.000000,15.000000,18.000000,16.000000,18.000000,21.000000,12.000000,13.500000,...,1.000000,0.000000,1.000000,0.000000,1.000000,0.000000,1.000000,2.000000,7.000000,1.000000
50%,40.000000,91.000000,53.000000,57.000000,62.000000,53.000000,56.000000,55.000000,46.000000,44.000000,...,8.000000,10.000000,6.000000,7.000000,9.000000,7.000000,8.000000,11.000000,31.000000,6.000000
75%,145.500000,429.000000,231.500000,267.000000,224.500000,221.000000,285.000000,272.500000,141.500000,162.500000,...,63.500000,77.500000,87.500000,85.500000,76.000000,70.000000,85.500000,80.000000,174.500000,70.500000
max,79055.000000,33364.000000,64027.000000,70720.000000,63361.000000,58636.000000,73520.000000,69225.000000,58080.000000,75592.000000,...,116037.000000,96191.000000,105413.000000,100387.000000,97278.000000,117400.000000,123705.000000,96470.000000,117867.000000,101379.000000


## Gut Microbiome Data Profiling

In this step, we loaded the gut microbiome abundance table (`microbiome.csv`) generated from the Johnson et al. 2019 dataset.

We performed an initial profiling of the dataset to better understand its structure and characteristics before preprocessing and filtering.

### What we examined
- Dataset dimensions
- Number of microbial features (taxa/species)
- Number of samples
- First rows of the abundance table
- Descriptive statistics for abundance values

### Results
- The dataset contains:
  - **223 microbial features**
  - **483 fecal samples**
- Rows represent microbial taxa/species.
- Columns represent individual microbiome samples.
- The abundance values are highly variable across samples, with many low values and zeros, which is common in microbiome datasets.

### Key observations
- Some bacterial species are highly abundant in specific samples.
- Many taxa appear at very low abundance levels.
- The dataset is expected to be sparse, meaning many entries contain zeros due to absent or rare microbes.

This profiling step provides the foundation for the next stages:
1. Low-prevalence feature filtering
2. Sparsity analysis
3. Data preprocessing for downstream microbiome analysis

# Analyzing Sparsity and Feature Prevalence 

In [6]:
# Calculate sparsity (percentage of zeros)

total_values = microbiome.size
zero_values = (microbiome == 0).sum().sum()

sparsity = (zero_values / total_values) * 100

print("Total values in dataset:", total_values)
print("Number of zero values:", zero_values)
print(f"Sparsity: {sparsity:.2f}%")

Total values in dataset: 107709
Number of zero values: 9467
Sparsity: 8.79%


In [7]:
# Prevalence of each microbial feature
# (fraction of samples where feature appears)

prevalence = (microbiome > 0).sum(axis=1) / microbiome.shape[1]

print(prevalence.describe())

# Show first prevalence values
display(prevalence.head())

count    223.000000
mean       0.912106
std        0.145638
min        0.271222
25%        0.910973
50%        0.973085
75%        0.994824
max        1.000000
dtype: float64


#taxonomy
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;NA                                   1.000000
k__Bacteria;p__Firmicutes;c__Clostridia;o__Clostridiales;f__Ruminococcaceae;g__Faecalibacterium;s__Faecalibacterium prausnitzii    1.000000
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Bacteroidaceae;g__Bacteroides;s__Bacteroides uniformis             1.000000
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Prevotellaceae;g__Prevotella;s__Prevotella copri                   0.910973
k__Bacteria;p__Bacteroidetes;c__Bacteroidia;o__Bacteroidales;f__Rikenellaceae;g__Alistipes;s__Alistipes putredinis                 0.993789
dtype: float64

## Sparsity and Prevalence Analysis

In this step, we analyzed the sparsity and prevalence of the gut microbiome abundance table.

### Sparsity Analysis
We calculated the percentage of zero values in the dataset to measure sparsity.

### Results
- Total values in the dataset: **107,709**
- Zero values: **9,467**
- Dataset sparsity: **8.79%**

This indicates that most microbial features are present in many samples, while a smaller portion of the matrix contains absent microbes (zeros).

### Prevalence Analysis
We then calculated the prevalence of each microbial feature, defined as the fraction of samples in which a feature appears.

### Key observations
- The average prevalence across taxa is very high (~0.91).
- Many microbial species appear in nearly all samples.
- Some taxa are less common and appear only in a subset of participants.
- The minimum prevalence observed was approximately **27%** of samples.

These results suggest that the dataset is relatively dense compared to many microbiome datasets, but still contains lower-prevalence taxa that may be filtered in the next preprocessing stage.

# Filtering Low-Prevalence Microbial Features

In [8]:
# Filter low-prevalence microbial features

threshold = 0.10   # keep taxa appearing in at least 10% of samples

filtered_microbiome = microbiome.loc[prevalence >= threshold]

print("Original shape:", microbiome.shape)
print("Filtered shape:", filtered_microbiome.shape)

print("Removed features:", microbiome.shape[0] - filtered_microbiome.shape[0])

Original shape: (223, 483)
Filtered shape: (223, 483)
Removed features: 0


In [9]:
filtered_microbiome.to_csv("microbiome_filtered.csv")

print("Filtered microbiome dataset saved.")

Filtered microbiome dataset saved.


In [10]:
import pandas as pd
import os

"""
paper2_data.py
Loads and exports datasets from Carlino et al. 2024 (Cell)
Source: https://github.com/SegataLab/cFMD
"""

# Base link to database
base_cfmd = "cFMD/cFMD_data"

# Dataset of microbial species  vs fecal samples
data_base = []
# Iterate on each dataset in the database
for dataset in os.listdir(base_cfmd):
    # Create path to dataset
    path = os.path.join(base_cfmd, dataset, f"{dataset}_taxonomic_profiles.tsv")
    # If the path is correct add the dataset to the data base
    if os.path.exists(path):
        df = pd.read_csv(path, sep="\t", index_col=0)
        data_base.append(df)

# Join the data sets to one single data set
combined = pd.concat(data_base, axis=1)
# Fill null value with zero
combined = combined.fillna(0)
print("Shape:", combined.shape)

# save the combined datasets to csv
combined.to_csv("combined_datasets.csv")

# Metadata on the microbiome
metadata = pd.read_csv("cFMD/cFMD_metadata.tsv", sep="\t")
# Export the metadata to csv
metadata.to_csv("cfmd_metadata.csv", index=False)

FileNotFoundError: [WinError 3] The system cannot find the path specified: 'cFMD/cFMD_data'

In [11]:
import os

for item in os.listdir(r"C:\Users\rtajz\Downloads"):
    if "cFMD" in item or "cfmd" in item:
        print(item)

# Loading and Combining the Paper 2 (cFMD) Microbiome Dataset

In [12]:
import pandas as pd
import os

# Paper 2: Carlino et al. 2024 / cFMD dataset
base_cfmd = r"C:\Users\rtajz\Downloads\cFMD-main\cFMD-main\cFMD_data"
metadata_path = r"C:\Users\rtajz\Downloads\cFMD-main\cFMD-main\cFMD_metadata.tsv"

print("Base exists:", os.path.exists(base_cfmd))
print("Metadata exists:", os.path.exists(metadata_path))

data_base = []

for dataset in os.listdir(base_cfmd):
    path = os.path.join(base_cfmd, dataset, f"{dataset}_taxonomic_profiles.tsv")
    
    if os.path.exists(path):
        df = pd.read_csv(path, sep="\t", index_col=0)
        data_base.append(df)

combined = pd.concat(data_base, axis=1)
combined = combined.fillna(0)

print("=== Paper 2 combined dataset ===")
print("Shape:", combined.shape)
display(combined.head())

combined.to_csv("combined_datasets.csv")

metadata = pd.read_csv(metadata_path, sep="\t")

print("=== Paper 2 metadata ===")
print("Shape:", metadata.shape)
display(metadata.head())

metadata.to_csv("cfmd_metadata.csv", index=False)

print("Done! Paper 2 CSV files were saved.")

Base exists: True
Metadata exists: True
=== Paper 2 combined dataset ===
Shape: (5057, 3393)


,AlmeidaO_2020__SRR8742576,AlmeidaO_2020__SRR8742577,AlmeidaO_2020__SRR8742578,AlmeidaO_2020__SRR8742579,AlmeidaO_2020__SRR8742580,AlmeidaO_2020__SRR8742581,AlmeidaO_2020__SRR8742582,AlmeidaO_2020__SRR8742584,AlmeidaO_2020__SRR8742585,AlvarezOrdonezA_xxxx__C01-19,...,YuY_2022__SRR16950700,YuY_2022__SRR16950701,YuY_2022__SRR16950702,YuY_2022__SRR16950703,YuY_2022__SRR16950704,ZhaoCC_2020__SRS5803191,ZhaoCC_2020__SRS5803192,ZhaoCC_2020__SRS5803193,ZhaoCC_2020__SRS5803194,ZhaoCC_2020__SRS5803195
sample,,,,,,,,,,,,,,,,,,,,,
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Streptococcaceae|g__Lactococcus|s__Lactococcus_lactis|t__SGB7985,0.14109,0.0,0.40403,0.00000,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,9.44002,9.00307,0.02590,0.02644,0.01921,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Actinomycetota|c__Actinomycetes|o__Propionibacteriales|f__Propionibacteriaceae|g__Cutibacterium|s__Cutibacterium_acnes|t__SGB16955,0.00000,0.0,0.00000,0.62919,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,0.00000,0.00000,0.00084,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Lactobacillaceae|g__Lactiplantibacillus|s__Lactiplantibacillus_plantarum|t__SGB7202,0.00000,0.0,0.00000,0.00000,0.24468,0.00653,0.08219,0.0,0.00553,0.0,...,51.21205,52.55470,0.12650,0.13905,0.12204,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Pseudomonadota|c__Gammaproteobacteria|o__Enterobacterales|f__Erwiniaceae|g__Pantoea|s__Pantoea_coffeiphila|t__SGB23192,0.10390,0.0,0.08975,0.00000,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Paenibacillaceae|g__Paenibacillus|s__Paenibacillus_cookii|t__SGB7404,0.00000,0.0,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.00236,0.0,...,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0


=== Paper 2 metadata ===
Shape: (3444, 32)


,dataset_name,sample_id,macrocategory,category,type,subtype,commercial_name,fermented/non-fermented,country,sample_accession,...,median_read_len,mean_read_len,max_read_len,n_contigs,n_MAGs_MQ_prok,n_MAGs_HQ_prok,n_MAGs_MQ_euk,n_MAGs_HQ_euk,filtered,curator
0,AlmeidaO_2020,SRR8742581,food,fermented_seeds,cocoa_beans,NaN,NaN,F,BRA,SAMN11159369,...,76.0,75.499143,76.0,467.0,0.0,1.0,0.0,0.0,no,Liam Walsh
1,AlmeidaO_2020,SRR8742582,food,fermented_seeds,cocoa_beans,NaN,NaN,F,BRA,SAMN11159368,...,76.0,75.510600,76.0,5182.0,2.0,0.0,1.0,0.0,no,Liam Walsh
2,AlmeidaO_2020,SRR8742584,food,fermented_seeds,cocoa_beans,NaN,NaN,F,BRA,SAMN11159372,...,76.0,75.504622,76.0,1719.0,1.0,1.0,0.0,0.0,no,Liam Walsh
3,AlmeidaO_2020,SRR8742585,food,fermented_seeds,cocoa_beans,NaN,NaN,F,BRA,SAMN11159371,...,76.0,75.506052,76.0,6344.0,0.0,2.0,0.0,0.0,no,Liam Walsh
4,AlmeidaO_2020,SRR8742578,food,fermented_seeds,cocoa_beans,NaN,NaN,F,BRA,SAMN11159364,...,76.0,75.564474,76.0,6514.0,2.0,0.0,1.0,0.0,no,Liam Walsh


Done! Paper 2 CSV files were saved.


## Paper 2 Dataset Integration and Exploration

In this step, we loaded and combined multiple microbiome datasets from the cFMD database (Carlino et al. 2024).

### What was done
- Loaded all taxonomic profile tables from the cFMD repository.
- Merged all datasets into one combined abundance matrix.
- Filled missing values with zeros.
- Loaded and explored the accompanying metadata table.
- Exported both the combined microbiome dataset and metadata to CSV files.

### Results
- Combined microbiome dataset shape:
  - **5057 microbial features**
  - **3393 samples**
- Metadata table shape:
  - **3444 samples**
  - **32 metadata variables**

### Key observations
- The combined dataset contains thousands of microbial taxa collected from many food-related microbiome samples.
- Missing values were replaced with zeros to create a complete abundance matrix for downstream analysis.
- The metadata includes detailed information such as:
  - dataset source
  - food category
  - fermentation status
  - country
  - sequencing information

This integrated dataset will be used for profiling, sparsity analysis, prevalence filtering, and food microbiome exposure analysis in later stages of the project.

# Profiling the Paper 2 Food Microbiome Dataset

In [13]:
import pandas as pd
import numpy as np

combined = pd.read_csv("combined_datasets.csv", index_col=0)

print("Shape:", combined.shape)
print("Number of microbial features:", combined.shape[0])
print("Number of samples:", combined.shape[1])

print("\nFirst rows:")
display(combined.head())

print("\nBasic statistics:")
display(combined.describe())

C:\Users\rtajz\AppData\Local\Temp\ipykernel_6276\1864880361.py:4: DtypeWarning: Columns (823) have mixed types. Specify dtype option on import or set low_memory=False.
  combined = pd.read_csv("combined_datasets.csv", index_col=0)


Shape: (5057, 3393)
Number of microbial features: 5057
Number of samples: 3393

First rows:


,AlmeidaO_2020__SRR8742576,AlmeidaO_2020__SRR8742577,AlmeidaO_2020__SRR8742578,AlmeidaO_2020__SRR8742579,AlmeidaO_2020__SRR8742580,AlmeidaO_2020__SRR8742581,AlmeidaO_2020__SRR8742582,AlmeidaO_2020__SRR8742584,AlmeidaO_2020__SRR8742585,AlvarezOrdonezA_xxxx__C01-19,...,YuY_2022__SRR16950700,YuY_2022__SRR16950701,YuY_2022__SRR16950702,YuY_2022__SRR16950703,YuY_2022__SRR16950704,ZhaoCC_2020__SRS5803191,ZhaoCC_2020__SRS5803192,ZhaoCC_2020__SRS5803193,ZhaoCC_2020__SRS5803194,ZhaoCC_2020__SRS5803195
sample,,,,,,,,,,,,,,,,,,,,,
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Streptococcaceae|g__Lactococcus|s__Lactococcus_lactis|t__SGB7985,0.14109,0.0,0.40403,0.00000,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,9.44002,9.00307,0.02590,0.02644,0.01921,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Actinomycetota|c__Actinomycetes|o__Propionibacteriales|f__Propionibacteriaceae|g__Cutibacterium|s__Cutibacterium_acnes|t__SGB16955,0.00000,0.0,0.00000,0.62919,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,0.00000,0.00000,0.00084,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Lactobacillaceae|g__Lactiplantibacillus|s__Lactiplantibacillus_plantarum|t__SGB7202,0.00000,0.0,0.00000,0.00000,0.24468,0.00653,0.08219,0.0,0.00553,0.0,...,51.21205,52.55470,0.12650,0.13905,0.12204,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Pseudomonadota|c__Gammaproteobacteria|o__Enterobacterales|f__Erwiniaceae|g__Pantoea|s__Pantoea_coffeiphila|t__SGB23192,0.10390,0.0,0.08975,0.00000,0.00000,0.00000,0.00000,0.0,0.00000,0.0,...,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0
k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Paenibacillaceae|g__Paenibacillus|s__Paenibacillus_cookii|t__SGB7404,0.00000,0.0,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.00236,0.0,...,0.00000,0.00000,0.00000,0.00000,0.00000,0.0,0.0,0.0,0.0,0.0



Basic statistics:


,AlmeidaO_2020__SRR8742576,AlmeidaO_2020__SRR8742577,AlmeidaO_2020__SRR8742578,AlmeidaO_2020__SRR8742579,AlmeidaO_2020__SRR8742580,AlmeidaO_2020__SRR8742581,AlmeidaO_2020__SRR8742582,AlmeidaO_2020__SRR8742584,AlmeidaO_2020__SRR8742585,AlvarezOrdonezA_xxxx__C01-19,...,YuY_2022__SRR16950700,YuY_2022__SRR16950701,YuY_2022__SRR16950702,YuY_2022__SRR16950703,YuY_2022__SRR16950704,ZhaoCC_2020__SRS5803191,ZhaoCC_2020__SRS5803192,ZhaoCC_2020__SRS5803193,ZhaoCC_2020__SRS5803194,ZhaoCC_2020__SRS5803195
count,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,...,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000,5057.000000
mean,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,...,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775,0.019775
std,0.464130,0.809432,0.702070,1.000120,0.823940,1.377910,1.207965,1.222904,0.989689,1.378808,...,0.790612,0.804994,0.589352,0.590576,0.609805,1.001798,1.018110,1.352622,0.754040,0.738988
min,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
25%,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
50%,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
75%,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,...,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000
max,14.899330,50.947280,34.887790,68.751090,45.144070,97.984570,85.627740,86.723460,68.564110,98.031220,...,51.212050,52.554700,31.997520,32.200410,35.128140,67.891230,69.383600,96.157660,44.609890,49.369390


## Profiling the Paper 2 Food Microbiome Dataset

In this step, we performed an initial exploration and profiling of the combined cFMD microbiome abundance dataset.

### What was examined
- Dataset dimensions
- Number of microbial features
- Number of samples
- First rows of the abundance matrix
- Descriptive statistics of microbial abundance values

### Results
- The dataset contains:
  - **5057 microbial features**
  - **3393 microbiome samples**
- Rows represent microbial taxa/species.
- Columns represent individual food microbiome samples collected from multiple datasets.

### Key observations
- The dataset is substantially larger than the Paper 1 microbiome dataset.
- Many abundance values are very small or zero, indicating the presence of rare microbial taxa.
- The abundance distributions are highly skewed, which is common in microbiome sequencing data.
- Some microbial taxa appear frequently across samples, while others are highly sparse.

This profiling step provides the basis for further preprocessing, including sparsity analysis, prevalence filtering, and downstream microbiome exposure analysis.

# Sparsity and Prevalence Analysis of the Paper 2 Dataset

In [14]:
# Calculate sparsity

total_values = combined.size
zero_values = (combined == 0).sum().sum()

sparsity = (zero_values / total_values) * 100

print("Total values in dataset:", total_values)
print("Number of zero values:", zero_values)
print(f"Sparsity: {sparsity:.2f}%")

Total values in dataset: 17158401
Number of zero values: 17063753
Sparsity: 99.45%


In [16]:
# Convert all values to numeric
combined_numeric = combined.apply(pd.to_numeric, errors='coerce')

# Replace NaN with zero
combined_numeric = combined_numeric.fillna(0)

# Calculate prevalence
prevalence = (combined_numeric > 0).sum(axis=1) / combined_numeric.shape[1]

print(prevalence.describe())

display(prevalence.head())

count    5057.000000
mean        0.005502
std         0.021082
min         0.000000
25%         0.000295
50%         0.000884
75%         0.002653
max         0.561745
dtype: float64


sample
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Streptococcaceae|g__Lactococcus|s__Lactococcus_lactis|t__SGB7985                          0.561745
k__Bacteria|p__Actinomycetota|c__Actinomycetes|o__Propionibacteriales|f__Propionibacteriaceae|g__Cutibacterium|s__Cutibacterium_acnes|t__SGB16955    0.071913
k__Bacteria|p__Firmicutes|c__Bacilli|o__Lactobacillales|f__Lactobacillaceae|g__Lactiplantibacillus|s__Lactiplantibacillus_plantarum|t__SGB7202       0.181256
k__Bacteria|p__Pseudomonadota|c__Gammaproteobacteria|o__Enterobacterales|f__Erwiniaceae|g__Pantoea|s__Pantoea_coffeiphila|t__SGB23192                0.007073
k__Bacteria|p__Firmicutes|c__Bacilli|o__Bacillales|f__Paenibacillaceae|g__Paenibacillus|s__Paenibacillus_cookii|t__SGB7404                           0.003831
dtype: float64

### Prevalence Analysis of the Paper 2 Dataset

We analyzed the prevalence of microbial features in the Paper 2 microbiome dataset after converting all values to numeric format. Since the dataset originally contained mixed data types, non-numeric values were converted to `NaN` and then replaced with zeros to enable numerical analysis.

The prevalence calculation measured the fraction of samples in which each microbial feature appeared (abundance > 0).

The results show that the dataset is highly sparse:
- The average prevalence is very low (~0.0055).
- Many microbial taxa appear in only a very small number of samples.
- Some taxa are present in more samples, with the maximum prevalence reaching ~56%.

These findings are consistent with large-scale microbiome datasets, where most microbial species are rare and only a subset are commonly observed across samples.